demos/scripts/robot_kinematics_control_loop.cpp

```diff
@@ -31,12 +31,12 @@ class DummyRobotInterface {
   void read_robot_state() {
     // this is a dummy robot, we assume the joint state is executed but add a bit of noise
     this->joint_state += 0.001 * JointState::Random(this->joint_state.get_name(), this->robot_.get_joint_frames());
-    this->eef_state = this->robot_.forward_geometry(this->joint_state);
+    this->eef_state = this->robot_.forward_kinematics(this->joint_state);
```

Comment:
This should be called with `JointPositions` and you would only get a twist back? Same comment for line 39

Reply:
Well yes and no. I can explicitly specify it for clarity but the call with a `State` is also perfectly fine as there is conversion happening there and we now are certain that those conversions only take the state variables they need after PR #100. Do you prefer that I change here just for clarity?

Reply:
I don't have a strong preference. Maybe in a demo, we should be super clear and not confuse a new user..